master/internal/stream/system.go

```diff
@@ -100,22 +93,36 @@ func NewPublisherSet() *PublisherSet {
 	}
 }
 
+// SyncMsg is the server response to a StartupMsg once it's been handled.
+type SyncMsg struct {
+	SyncID    string `json:"sync_id"`
+	ErrString string `json:"error"`
```

Comment:
Why do we need to introduce an additional in-band error message here?  What would the client do with an error here if they saw one?

Reply:
I added this to:
- Inform the streaming client of startup failure types (malformed startup message vs. something like a database failure).

However on second thought:
- The startup message case is resolved by validation on the streaming client side and logging on the server side, which is already being done.
  - Logging on the server side aids debugging even if validation between the streaming client and master fails.
- The client response to a database failure should be consistent: see websocket close and attempt to reconnect.
  
One thing that this does help with is letting streaming client users know why there was an (unrecoverable) failure, in the event they don't have access to the logs.

Reply:
Sounds good, I'm of the mindset to remove it for now